application/electron/src/services/files/service.file.opener.ts

```diff
@@ -228,6 +256,94 @@ class ServiceFileOpener implements IService {
+                            ServiceNotifications.notify({
+                                caption: 'Error with file',
+                                message: errorMessage,
+                                type: ENotificationType.warning,
+                            });
+                            return rejected(new Error(errorMessage));
+                        }
+                        if (lsStats.isFile()) {
+                            // File
+                            return fs.stat(file, (fsErr, fsStats) => {
+                                if (fsErr) {
+                                    const errorMessage = `Fail to get file info of ${file} due to error: ${fsErr.message}`;
+                                    ServiceNotifications.notify({
+                                        caption: 'Error with file',
+                                        message: errorMessage,
+                                        type: ENotificationType.warning,
+                                    })
+                                    rejected(new Error(errorMessage));
+                                } else {
+                                    resolved(allFiles.push({
+                                        lastModified: fsStats.mtimeMs,
+                                        lastModifiedDate: fsStats.mtime,
+                                        name: self._getName(file),
+                                        path: file,
+                                        size: fsStats.size,
+                                        type: 'file',
+                                    }));
+                                }
+                            });
+                        } else if (!(lsStats.isDirectory())) {
+                            // Neither file nor directory
+                            const errorMessage = `Fail to get file info of ${file} because it is neither a file nor a directory`;
+                            ServiceNotifications.notify({
+                                caption: `Error with file`,
+                                message: errorMessage,
+                                type: ENotificationType.warning,
+                            });
+                            rejected(new Error(errorMessage));
+                        } else {
+                            // Directory
+                            return fs.readdir(file, (err, files) => {
+                                if (err) {
+                                    const errorMessage = `Fail to list files of directory ${file} due to error: ${err.message}`;
+                                    ServiceNotifications.notify({
+                                        caption: 'Error with directory',
+                                        message: errorMessage,
+                                        type: ENotificationType.warning,
+                                    });
```

Comment:
We cannot use notifications here because:
- In the case of a request from front-end (render) notifications could be used. Render could show it by itself
- Task of service (in our case IPC request): build and return list of file. Only this. We should not do any other actions, which weren't required by IPC request.